app/models/miq_product_feature.rb

```diff
@@ -77,50 +75,59 @@ def self.seed
   end
 
   def self.seed_features
-    idents_from_hash = []
-    self.seed_from_hash(YAML.load_file(FIXTURE_YAML), idents_from_hash)
+    seen = Hash.new([])
+    features = all.to_a
+    self.seed_from_hash(YAML.load_file(FIXTURE_YAML), seen, nil, features)
 
-    root_feature = MiqProductFeature.where(:identifier => 'everything').first
-    Dir.glob(File.join(FIXTURE_PATH, "*.yml")).each do |fixture|
-      self.seed_from_hash(YAML.load_file(fixture), idents_from_hash, root_feature)
-    end
-
-    idents_from_db = self.all.collect(&:identifier)
-    deletes = idents_from_db - (idents_from_db & idents_from_hash)
-    unless deletes.empty?
-      _log.info("Deleting product features: #{deletes.inspect}")
-      self.destroy_all(:identifier => deletes)
-    end
+    deletes = where.not(:identifier => seen.values.flatten).destroy_all
+    _log.info("Deleting product features: #{deletes.collect(&:identifier).inspect}") unless deletes.empty?
+    seen
   end
 
-  def self.seed_from_hash(hash, seen = [], parent=nil)
+  def self.seed_from_hash(hash, seen = {}, parent = nil, features = nil)
     children = hash.delete(:children) || []
     hash.delete(:parent_identifier)
 
     hash[:parent] = parent
-    feature = self.find_by_identifier(hash[:identifier])
+
+    db_feature =
+      if features
+        features.detect { |f| f.identifier == hash[:identifier] }
+      else
+        where(:identifier => hash[:identifier]).first
```

Comment:
Yes, I pass it in to eliminate this N+1. The only place we don't is for seed specific features in the tests.  Normal seeding will pass in all of the features up front instead of querying them here.